DependencyInjection/GttCryptExtension.php

```diff
@@ -95,15 +98,72 @@ protected function registerRsaCryptor($name, $cryptorConfig, Definition $registr
         $rsaDecryptorDefinition = new DefinitionDecorator('gtt.crypt.rsa.decryptor');
         $rsaDecryptorDefinition->replaceArgument(0, $zendRsaReference);
 
-        $rsaEncryptorDefinitionId = CryptorServiceIdGenerator::generateEncryptorId($name);
-        $rsaDecryptorDefinitionId = CryptorServiceIdGenerator::generateDecryptorId($name);
+        $this->setCryptorsPair(
+            $name,
+            $container,
+            $registryDefinition,
+            $rsaEncryptorDefinition,
+            $rsaDecryptorDefinition
+        );
+    }
+
+    /**
+     * Register the symmetric cryptors in container
+     *
+     * @param string           $name                cryptor name
+     * @param array            $cryptorConfig       cryptor config
+     * @param Definition       $registryDefinition  registry service definition
+     * @param ContainerBuilder $container           container
+     */
+    protected function registerSymmetricCryptor($name, $cryptorConfig, Definition $registryDefinition, ContainerBuilder $container)
+    {
+        $keyReaderDefinition = new DefinitionDecorator('gtt.crypt.symmetric.key_reader');
+        $keyReaderDefinition->replaceArgument(0, $cryptorConfig['path']);
+        $keyReaderDefinition->setPublic(false);
+        $keyReaderDefinitionId = "gtt.crypt.symmetric.key_reader.$name";
+        $container->setDefinition($keyReaderDefinitionId, $keyReaderDefinition);
+        $keyReaderReference = new Reference($keyReaderDefinitionId);
+
+        $symmetricEncryptorDefinition = new DefinitionDecorator('gtt.crypt.symmetric.encryptor');
+        $symmetricEncryptorDefinition->setArguments(array($keyReaderReference, $cryptorConfig['base64']));
+
+        $symmetricDecryptorDefinition = new DefinitionDecorator('gtt.crypt.symmetric.decryptor');
+        $symmetricDecryptorDefinition->setArguments(array($keyReaderReference, $cryptorConfig['base64']));
+
+        $this->setCryptorsPair(
+            $name,
+            $container,
+            $registryDefinition,
+            $symmetricEncryptorDefinition,
+            $symmetricDecryptorDefinition
+        );
+    }
+
+    /**
+     * Add encryptor and decryptor definitions to container and cryptor registry
+     *
+     * @param string           $name                Cryptor name
+     * @param ContainerBuilder $container           Container
+     * @param Definition       $registryDefinition  Registry definition
+     * @param Definition       $encryptorDefinition Encryptor definition
+     * @param Definition       $decryptorDefinition Decryptor definition
+     */
+    private function setCryptorsPair(
```

Comment:
It would be great to refactor the registerRsaCryptor method and use setCryptoesPair method inside it